R/combine_checkboxes.R

```diff
@@ -0,0 +1,192 @@
+#'   values_to = "race_combined",
+#'   multi_value_label = "Multiple",
+#'   values_fill = NA
+#' )
+#' }
+#'
+#' @export
+
+combine_checkboxes <- function(supertbl,
+                               tbl,
+                               cols,
+                               values_to,
+                               multi_value_label = "Multiple",
+                               values_fill = NA,
+                               raw_or_label = "label",
+                               keep = TRUE) {
+
+  # Check args ---
+  check_arg_is_supertbl(supertbl, req_cols = c("redcap_data", "redcap_metadata"))
+  check_arg_is_character(tbl, len = 1, any.missing = FALSE)
+  check_arg_is_character(values_to, len = 1, any.missing = FALSE)
+  check_arg_is_character(multi_value_label, len = 1, any.missing = TRUE)
+  check_arg_is_character(values_fill, len = 1, any.missing = TRUE)
+  check_arg_choices(raw_or_label, choices = c("label", "raw"))
+  check_arg_is_logical(keep, len = 1, any.missing = FALSE)
+
+  # Save user cols to enquosure
+  cols_exp <- enquo(cols)
+
+  # Extract tbl from supertbl
+  data_tbl <- supertbl %>%
+    extract_tibble(tbl)
+
+  # Get field names from cols_exp, check that fields exist
+  field_names <- names(eval_select(cols_exp, data = data_tbl))
+  check_fields_exist(fields = field_names, expr = cols_exp)
+
+  # Assume the first instrument in the metadata contains IDs
+  # REDCap enforces this constraints, we reflect this in read_redcap -> get_field_to_drop
+  record_id_field <- supertbl$redcap_metadata[supertbl$redcap_form_name == tbl][[1]]$field_name[1]
+
+  # Combine record identifier with remaining possible project identifiers
+  instrument_identifiers <- c(
+    record_id_field,
+    "redcap_form_instance",
+    "redcap_form_name",
+    "redcap_event",
+    "redcap_event_instance"
+  )
+
+  # Define values_to as the count of TRUEs/1s for the given checkbox field
+  # Assign TRUE if multiple selections made, and FALSE if one or zero made
+  data_tbl_mod <- data_tbl %>%
+    select(any_of(instrument_identifiers), !!!eval_select(cols_exp, data_tbl)) %>%
+    mutate(
+      !!values_to := case_when(
+        rowSums(select(., eval_tidy(cols_exp))) > 1 ~ TRUE,
+        TRUE ~ FALSE
+      )
+    )
+
+  # Get metadata reference table, check that chosen fields are checkboxes
+  metadata <- get_metadata_ref(data_tbl_mod, supertbl, tbl, instrument_identifiers)
+
+  # Replace TRUEs/1s with raw/label values from metadata
+  data_tbl_mod <- data_tbl_mod %>%
+    mutate(across(
+      -c(any_of(instrument_identifiers), !!values_to),
+      ~ replace_true(.x,
+        cur_column(),
+        metadata = metadata,
+        raw_or_label = raw_or_label
+      )
+    ))
+
+  # Convert values_to from TRUE/FALSE to multi_value_label or identified single val
+  data_tbl_mod <- data_tbl_mod %>%
+    mutate(
+      across(field_names, as.character), # enforce to character strings
+      across(!!values_to, ~ as.character(.))
```

Comment:
What's the need to convert this?

Reply:
After the transformations before it, it's possible for checked columns to have character string labels but unchecked ones to still come up as `lgl`. See cols `race___3`, `race___4`, `race___5`:

```r
data_tbl_mod
# A tibble: 3 × 10
  study_id redcap_event race___0                      race___1 race___2                                  race___3 race___4 race___5 race___6 new_race_col
     <dbl> <chr>        <chr>                         <chr>    <chr>                                     <lgl>    <lgl>    <lgl>    <lgl>    <lgl>       
1        1 event_1      NA                            Asian    NA                                        NA       NA       NA       NA       FALSE       
2        2 event_1      American Indian/Alaska Native NA       Native Hawaiian or Other Pacific Islander NA       NA       NA       NA       TRUE        
3        3 event_1      American Indian/Alaska Native Asian    NA                                        NA       NA       NA       NA       TRUE  
```

Reply:
Ah yes I see, and then just make `ifelse(!!sym(values_to)...`. Roger roger.